[DEMO] Smartlab #demo (#3058)
openvinotoolkit/open_model_zoo: demos/smartlab_online_demo/python/README.md

# Smartlab online Python\* Demo

> This is the demo applicatio with smartlab action recognition and smartlab object detection algorithms.

Reviewer: Can you first introduce `smartlab` for readers?

Reviewer: I can't see it

> This is the demo applicatio with smartlab action recognition and smartlab object detection algorithms.

Reviewer: ```suggestion
This is the demo application with smartlab action recognition and smartlab object detection algorithms.
```

Author: fixed

The following pre-trained models are delived with the product:

> `

Reviewer: ```suggestion
```

Author: removed

> * `smartlab-action-recognition-encoder-0001` + `smartlab-action-recognition-decoder-0001`, which are models for identifying actions of smartlab (2 actions). They recognize actions like adjust rider, put_take.

Reviewer: ```suggestion
* `smartlab-action-recognition-encoder-0001` + `smartlab-action-recognition-decoder-0001`, which are models for identifying 2 actions of smartlab (adjust_rider, put_take).
```

Reviewer: You have double `smartlab` now:
> smartlab 2 actions of smartlab

Author: thanks. removed the  redundant word

> * `smartlab-object-detection-0001` + `smartlab-object-detection-0002` + `smartlab-object-detection-0003` + `smartlab-object-detection-0004`, which are models for detecing smartlab objectsi (10 objects). They detect balance, weights, tweezers, box, battery, tray, ruler, rider, scale, hand.

Reviewer: ```suggestion
* `smartlab-object-detection-0001` + `smartlab-object-detection-0002` + `smartlab-object-detection-0003` + `smartlab-object-detection-0004`, which are models for detecting smartlab objects (10 objects). They detect balance, weights, tweezers, box, battery, tray, ruler, rider, scale, hand.
```

Author: fixed

Author: fixed

> For more information about the pre-trained models, refer to the [Intel](../../../models/intel/index.md) and [public](../../../models/public/index.md) models documentation
> 

Reviewer: ```suggestion
```

Author: remvoed

## How It works

The demo pipeline consists of several steps, namely 'Decode', 'Detector', 'Segmentor', 'Evaluator', 'Display'.

> Every step is implemeted by correcponding class. Decode is implemented by opencv API, Detector is imlemeted `detector.py`. Segmentor is implemented by `segmentor.py`. Evaluator is implemented by `segmentor.py`. Evaluator is implemented by `evaluator.py`. Display is implemneted by `display.py`

Reviewer: ```suggestion
Every step is implemented in corresponding class.
```

Author: fixed

Reviewer: I don't think you need this
> Every step is implemeted in correcponding class. Decode is implemented by opencv API, Detector is imlemeted `detector.py`. Segmentor is implemented by `segmentor.py`. Evaluator is implemented by `segmentor.py`. Evaluator is implemented by `evaluator.py`. Display is implemneted by `display.py`

You have good naming. And files names by themselves correspond to classes. So everything should be clear without explanation.

Author: removed

* `Decode` reads frames from the input videos.
* `Detector` detects smartlab objects (balance, weights, tweezers, box, battery, tray, ruler, rider scale, hand)

> * `Segmentor` segment video frames based on action of the frame.

Author: fixed

* `Evaluator` calculates scores of the current state.

> * `Display` display detected objects, regnized action, calculated scores on the current frame.

Author: fixed

> **NOTE**: By default, Open Model Zoo demos expect input with BGR channels order. If you trained your model to work with RGB order, you need to manually rearrange the default channels order in the demo application or reconvert your model using the Model Optimizer tool with the `--reverse_input_channels` argument specified. For more information about the argument, refer to **When to Reverse Input Channels** section of [Converting a Model Using General Conversion Parameters](https://docs.openvino.ai/latest/openvino_docs_MO_DG_prepare_model_convert_model_Converting_Model.html#general-conversion-parameters).

## Preparing to Run

> For demo input image or video files, need to provide smartlab videos following our setup (any shareable URL???).

Reviewer: You haven't filled `any shareable URL???`

Author: let me pending this issue. our dataset is not sharable.

Author: added demo video URL

An example of using the Model Downloader:

```sh
omz_downloader --list models.lst
```

An example of using the Model Converter:

```sh
omz_converter --list models.lst
```

### Supported Models
smartlab-action-recognition-encoder-0001
smartlab-action-recognition-decoder-0001
smartlab-object-detection-0001
smartlab-object-detection-0002
smartlab-object-detection-0003
smartlab-object-detection-0004

> **NOTE**: Refer to the tables [Intel's Pre-Trained Models Device Support](../../../models/intel/device_support.md) and [Public Pre-Trained Models Device Support](../../../models/public/device_support.md) for the details on models inference support at different devices.

## Running

Running the application with the `-h` option yields the following usage message:

```
usage: video_processor_serial.py [-h] -tv TOPVIEW -fv FRONTVIEW -m_ta M_TOPALL -m_tm M_TOPMOVE -m_fa M_FRONTALL -m_fm M_FRONTMOVE {multiview,mstcn} ...

positional arguments:
  {multiview,mstcn}     sub-command help
    multiview           multiview help

>     mstcn               mstcm help

Reviewer: I think think you didn't intend this to be an arg

Author: Actually we have two type demos (multiview and mstcn) in the demo application.
Is it better to separate them?

Currently, I removed mstcn.
